config/config.go

```diff
@@ -126,6 +128,15 @@ type BaseConfig struct {
 	RegossipAtxInterval time.Duration `mapstructure:"regossip-atx-interval"`
 }
 
+type deprecatedPoETServers struct{}
+
+func (deprecatedPoETServers) UnmarshalText([]byte) error {
```

Comment:
It's used by `mapstructure`. The idea is that when `post-server` is found in the config, then `mapstructure` will try to unmarshal it with this function and return an error.

Reply:
A custom hook like this instead of `UnmarshalText` implementation would work too. However, this hook could not use a logger because loggers are not yet available when it's used. It could use a global logger or an `fmt.Printf()`.

Reply:
With the hook being a closure an instance to the logger can be passed into it (see my code example). There is no need for a global logger 🙂

Reply:
nvm just saw that i was looking at an older state of the PR.